### PR TITLE
Allow for specifying protocol and API base path

### PR DESCRIPTION
**Pull request by bellpeterm**
I work in an environment where the ElasticSearch API is using HTTPS and its base path is /elasticsearch so both of these improvements are needed.  I thought it might be useful for others too.

### CODE REVIEW

File: src/main/java/me/snov/newrelic/elasticsearch/parsers/NodesStatsParser.java [hunk 1]
```diff
@@ -13,7 +13,7 @@ public NodesStatsParser() {
         super(NodesStats.class, null, null, null);
     }
 
-    public NodesStatsParser(String host, int port, String username, String password) throws MalformedURLException {
-        super(NodesStats.class, new URL(HTTP, host, port, URL_CLUSTER_STATS), username, password);
+    public NodesStatsParser(String protocol, String host, int port, String basePath, String username, String password) throws MalformedURLException {
+        super(NodesStats.class, new URL(protocol, host, port, basePath + URL_CLUSTER_STATS), username, password);
```
Comment (s12v) [inline src/main/java/me/snov/newrelic/elasticsearch/parsers/NodesStatsParser.java:17]: please cover a case, when the path ends with `/`?

File: src/main/java/me/snov/newrelic/elasticsearch/ElasticsearchAgentFactory.java [hunk 2]
```diff
@@ -26,13 +28,21 @@ public Agent createConfiguredAgent(Map<String, Object> properties) throws Config
             throw new ConfigurationException("'host' and 'port' must be specified. Do you have a 'config/plugin.json' file?");
         }
 
+        if (protocol == null) {
```
Comment (s12v) [inline src/main/java/me/snov/newrelic/elasticsearch/ElasticsearchAgentFactory.java:31]: could you please add the new options to the example configuration?

**Review by s12v — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks good, 2 minor comments and 👍

**Comment by bellpeterm** [comment]
Great feedback.  Updated PR.

Added case to drop '/' if it is last character in basePath.
Added documentation for basePath option in line with other optional parameter.

**Comment by s12v** [comment]
@bellpeterm, thank you. Could you please also add "protocol" to the example config?

**Comment by bellpeterm** [comment]
@s12v my mistake.  Updated the config again in the pull request.  Cheers!

**Review by s12v — APPROVED** [review]

**Comment by s12v** [comment]
thank you @bellpeterm

**Pull request merged**